framework/src/base/FEProblem.C

```diff
@@ -1252,7 +1250,7 @@ FEProblem::addKernel(const std::string & kernel_name, const std::string & name,
   parameters.set<FEProblem *>("_fe_problem") = this;
   if (_displaced_problem != NULL && parameters.get<bool>("use_displaced_mesh"))
   {
-    parameters.set<SubProblem *>("_subproblem") = _displaced_problem;
+    parameters.set<SubProblem *>("_subproblem") = _displaced_problem.get();
```

Comment:
Yeah - I don't think we can... just one more reason not to like SharedPointers ;-)